app/src/main/java/com/hamdan/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -19,182 +20,223 @@ class LoginViewModel @Inject constructor(
     private val requestValidationCode: LoginRequestValidationUseCase,
     private val createAccountRequestUseCase: CreateAccountRequestUseCase
 ) : ViewModel() {
-
-    sealed interface LoginState {
-        object Error : LoginState
-
-        data class Loading(
-            val email: String = "",
-            val emailError: Boolean = true,
-        ) : LoginState
-
-        data class UserInputting(
-            val email: String = "",
-            val emailError: Boolean = true,
-        ) : LoginState
-    }
-
-    sealed interface CreateAccountState {
-        object Loading : CreateAccountState
-        object Error : CreateAccountState
-
-        data class UserInputting(
-            val firstName: String = "",
-            val lastName: String = "",
-            val birthDay: String = "",
-            val email: String = "",
-            val location: String = "",
-            val firstError: Boolean = true,
-            val lastError: Boolean = true,
-            val birthError: Boolean = true,
-            val emailError: Boolean = true,
-            val locationError: Boolean = true,
-        ) : CreateAccountState
-    }
+    data class LoginState(
+        val email: Entry = Entry("", true),
+        val code: Entry = Entry("", true),
+        val showInfoDialog: Boolean = false,
+        val inputtingCode: Boolean = false,
+        val isLoading: Boolean = false,
+        val hasError: Boolean = false,
+    )
+
+    data class CreateAccountState(
+        val errorId: Int? = null,
+        val firstName: Entry = Entry("", true),
+        val lastName: Entry = Entry("", true),
+        val birthDay: Entry = Entry("", true),
+        val email: Entry = Entry("", true),
+        val location: Entry = Entry("", true),
+        val isLoading: Boolean = false,
+    )
 
     private var _createAccountState: MutableState<CreateAccountState> =
-        mutableStateOf(CreateAccountState.UserInputting())
+        mutableStateOf(CreateAccountState())
     val createAccountState: MutableState<CreateAccountState>
         get() = _createAccountState
 
     private var _loginState: MutableState<LoginState> =
-        mutableStateOf(LoginState.UserInputting())
+        mutableStateOf(LoginState(email = Entry("Hamdan.mobeen@Gmail.com", false)))
```

Comment:
You better put a TODO to remove this

Reply:
I had actually meant to remove this, and litteraly thought 3 seperate times ok time to remove it, then forgot to remove it each time.

Reply:
added a TODO for now